code/go/internal/validator/common/helpers.go

```diff
@@ -0,0 +1,44 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package common
+
+import (
+	"os"
+	"strconv"
+)
+
+// EnvVarWarningsAsErrors is the environment variable name used to use warnings as errors
```

Comment:
Please add a comment here mentioning that this mechanism will be removed if/when structured errors are supported.